### PR TITLE
16634 Base changes for "way of navigating" story (new branch "release-5.0.0")

### PR DESCRIPTION
**Pull request by severinbeauvais**
*Issue #:* /bcgov/entity#16634

**The "way of navigating" story (16519) contains many tickets that cannot be merged to main branch without breaking things. Hence, THIS pull request implements some base changes, including a new feature branch, for those changes. When the feature branch is working correctly, then it will be merged into main branch.**

*Description of changes:*
Commit 1:
    - app version = 5.0.0
    - used latest node version
    - updated browserslist
    - fixed a lint warning
    - created mixin-tester.vue
    - fixed date-mixin unit test errors
    - added babel plugin
    - updated axios, core-js, etc
    - updated types, test utils, etc
    - updated ts config
    - added volar config
    - changed some imports that gave errors
    - removed some "private" modifiers
    - added components.d.ts
    - deleted shims-tsx.d.ts
    - added vue-test-utils-helpers.d.ts
    - removed extra Vue.use()
    - misc cleanup

Commit 2:
    - moved mixin-tester.vue from src/ to tests/
    - moved interceptors.ts from sbc-common to local plugins to fix Axios version issue
    - other strict:true fixes
    - fixed property visibility issue
    - fixed catch clause variable issues
    - capitalized some plugin functions (eg, navigate -> Navigate)
    - fixed some visibility issues
    - misc cleanup

Commit 3:
    - fixed whitespace around v-rows
    - fixed empty object error
    - added misc code comments
    - changed some more class property/method visibilities

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the namerequest license (Apache 2.0).

### CODE REVIEW

**Comment by severinbeauvais** [comment]
Eve, Omid -- FYI only.

This is for [story 16519](https://app.zenhub.com/workspaces/entities-team-space-6143567664fb320019b81f39/issues/gh/bcgov/entity/16519).

Hopefully there won't be too many (or any) fixes to main branch while we develop on release-5.0.0 branch (will we will merge when it's safe to do so without breaking anything).

**Review by eve-git — APPROVED** [review]
Only the app version updated. 
Looks good to me.

**Comment by severinbeauvais** [comment]
> Only the app version updated. Looks good to me.

For now, yes. Should I merge this and then create more PRs for the rest of the work in this ticket, or can I add extra work to THIS PR? (In case you don't want to see the extra work -- do you?)

**Comment by eve-git** [comment]
I agree that it is better to build a base first.

**Comment by severinbeauvais** [comment]
This repo has unit test errors before the present changes.

Using Node 12, before changes:

image

Using Node 16, after changes:

image

**Comment by seeker25** [comment]
Your package-lock has changed quite a bit - I'd recommend an entire full retest

File: src/mixin-tester.vue [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  <div class="d-none" />
+</template>
+
+<script lang="ts">
+import { Component, Mixins } from 'vue-property-decorator'
+import { DateMixin } from '@/mixins/date-mixin'
+
+/** Special component used only for testing mixins. */
+@Component({})
+export default class MixinTester extends Mixins(DateMixin) {}
```
Comment (seeker25) [inline src/mixin-tester.vue:11]: Shouldn't this live in TESTS? or is that not possible?

**Review by seeker25 — APPROVED** [review]

File: src/mixin-tester.vue (hunk 1 above)
Comment (seeker25) [reply src/mixin-tester.vue:11]: 21 unit tests seems fairly low as well lol.

**Comment by severinbeauvais** [comment]
> Your package-lock has changed quite a bit - I'd recommend a decent retest.

For sure. We will be making a lot of changes to this feature branch so it will undergo lots of testing.

File: src/mixin-tester.vue (hunk 1 above)
Comment (severinbeauvais) [reply src/mixin-tester.vue]: Good call. Fixed.

**Review by vysakh-menon-aot — APPROVED** [review]

**Review by JazzarKarim — APPROVED** [review]

**Pull request merged**